components/templates/src/global_fns/load_data.rs

```diff
@@ -0,0 +1,346 @@
+extern crate toml;
+extern crate serde_json;
+
+use utils::fs::{read_file, is_path_in_directory, get_file_time};
+
+use crypto_hash::{Algorithm, hex_digest};
+use chrono::{DateTime, Utc};
+use reqwest::{Client, header};
+use url::Url;
+
+use std::path::PathBuf;
+use std::sync::{Arc, Mutex};
+
+
+use csv::Reader;
+use std::collections::HashMap;
+use tera::{GlobalFn, Value, from_value, to_value, Result, Map};
+use std::ops::BitXor;
+
+static GET_DATA_ARGUMENT_ERROR_MESSAGE: &str = "`load_data`: requires EITHER a `path` or `url` argument";
+
+enum ProvidedArgument {
+    URL(Url),
+    PATH(PathBuf)
```

Comment:
enum variants should be lowercase, `Url` and `Path`.
The enum name would be probably clearer as `DataSource` as well